### PR TITLE
update helpdesk logic to use Jira personal access tokens

### PR DESCRIPTION
**Pull request by justincorrigible**
The ARGO platform contact form was borked by changes in auth management at the Helpdesk API.
Our Jira server's Basic auth now requires captchas, etc. so this app hasn't been able to submit new tix since summer 2023.

This PR changes the way we do auth to use Jira PATs (same user as before, see credentials in prod vault).
It also improves the error handling for that section of the app, so the pod doesn't crash if Jira becomes unreachable for a different reason, or similar issues arise later on.

Along the way, I've added a couple goodies:
- improved error handling to reduce crashes in other areas (captcha, eslint clients)
- heaps of logging (including debug mode #rejoice)
- add ‘@/‘ to absolute paths to match current community/industry practices.

### CODE REVIEW

File: .env.schema [hunk 1]
```diff
@@ -1,5 +1,6 @@
 PORT=9000
 ADVERTISED_HOST=http://localhost:9000
+DEBUG=false # turns on `debug` level logging in production
```
Comment (justincorrigible) [inline .env.schema:3]: logging ftw

File: src/schemas/Helpdesk/jiraRequests.ts
Comment (justincorrigible) [inline src/schemas/Helpdesk/jiraRequests.ts:1]: summary of changes in this file (hard to see through all the prettier changes):
- change basic auth logic to use tokens instead
- abstracted helper functions from main function
- improved error handling, logging etc.

File: src/schemas/SystemAlert/index.ts
Comment (justincorrigible) [inline src/schemas/SystemAlert/index.ts:1]: summary, as in the previous file:
- hella prettier
- logging and error handling

File: src/services/elasticsearch/index.ts
Comment (justincorrigible) [inline src/services/elasticsearch/index.ts:1]: same

File: src/services/reCaptcha/index.ts
Comment (justincorrigible) [inline src/services/reCaptcha/index.ts:1]: does anyone else hear an echo?

File: src/services/vault/index.ts [hunk 2]
```diff
@@ -49,9 +50,11 @@ export const createVaultClient = memoize(async (vaultOptions: VaultOptions = {})
 
 export const loadVaultSecret =
 	(vaultClient: ReturnType<typeof createVaultClient> = createVaultClient()) =>
-	async (path: string) => {
+	async <T = Record<string, string>>(path: string): Promise<T> => {
```
Comment (justincorrigible) [inline src/services/vault/index.ts:53]: started something, didn't really finish it cognisant of scope, but should enable vault secret validation in the future

File: tsconfig.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
 		"outDir": "./dist",
 		"baseUrl": "./",
 		"paths": {
-			"*": ["./src/*"]
+			"*": ["./src/*"],
+			"@/*": ["./src/*"]
```
Comment (justincorrigible) [inline tsconfig.json:18]: gradual community practices adoption

File: tsconfig.eslint.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+	"compilerOptions": {
+		"allowJs": true
+	},
+	"exclude": ["dist/**/*", "src/**/*"],
+	"include": [".eslintrc.js", "prettier.config.js", "**/*.json"]
```
Comment (justincorrigible) [inline tsconfig.eslint.json:6]: validates eslint, prettier, and eventually arranger configs

File: src/utils/logger.ts [hunk 5]
```diff
@@ -17,21 +17,21 @@
...
 	format: combine(
 		timestamp(),
-		printf((info) => `${info.timestamp} ${info.level}: ${info.message}`),
+		printf((info) => `${info.timestamp} ${info.level}:\n  ${info.message}`),
 	),
 	transports: [
 		new transports.Console({
 			handleExceptions: true,
-			level: isProd ? 'error' : 'debug',
+			level: DEBUG || !IS_PROD ? 'debug' : 'info',
```
Comment (justincorrigible) [inline src/utils/logger.ts:34]: moar logs!1 #l33t

**Review by justincorrigible — COMMENTED** [review]
des comments for great good 💰

**Review by joneubank — APPROVED** [review]

**Pull request merged**